src/test/java/uk/gov/pay/card/bin_ranges/BinRangeTest.java

```diff
@@ -0,0 +1,134 @@
+        checkForBinRangeOverlaps("Discover", discoverData, "Test Card", testCardData);
+    }
+
+    @Test
+    void shouldHaveNoOverlapBetweenDiscoverAndWorldpayBinRanges() {
+        checkForBinRangeOverlaps("Worldpay", worldpayData, "Discover", discoverData);
+    }
+
+    private void checkForBinRangeOverlaps(String ranges1Type, List<Range<Long>> ranges1, String ranges2Type, List<Range<Long>> ranges2) {
+        RangeSet<Long> rangeSet = TreeRangeSet.create();
+        ranges1.forEach(rangeSet::add);
+
+        List<String> overlappingRanges = new ArrayList<>();
+        ranges2.forEach(cardRange -> {
+            if (rangeSet.intersects(cardRange)) {
+                overlappingRanges.add(String.format("%s range %s intersects %s ranges %s", 
+                        ranges1Type, cardRange, ranges2Type, rangeSet.subRangeSet(cardRange)));
+            }
+        });
+
+        if (!overlappingRanges.isEmpty()) {
+            fail("Found overlaps in ranges:\n" + String.join("\n" , overlappingRanges));
+        }
+    }
+
+    /**
+      We need to check there is no overlap between ranges across all data sources because some ranges are associated
+      with corporate card numbers, as identified by the 4th "CP" column in the example line entry from the Worldpay data source:
+
+      01,511226661120000000,511226661121000000,CP,DINERS DISCOVER,.... etc
+
+      Having no overlap ranges is essential to cardid figuring out with confidence if a card is a corporate card or not.
+     */
+    @Test
+    void verifyNoOverlappingRanges() {
+        RangeSet<Long> setOfBinRanges = TreeRangeSet.create();
+        Map<Range<Long>, String> mapOfRanges = new HashMap<>();
+        List<Range<Long>> rangesNotInHashMap = new ArrayList<>();
+        worldpayData.forEach(cardRange -> {
+            setOfBinRanges.add(cardRange);
+            mapOfRanges.put(cardRange, "worldpay");
+        });
+        discoverData.forEach(cardRange -> {
+            setOfBinRanges.add(cardRange);
+            mapOfRanges.put(cardRange, "discover");
+        });
+        testCardData.forEach(cardRange -> {
+            setOfBinRanges.add(cardRange);
+            mapOfRanges.put(cardRange, "test");
+        });
+        setOfBinRanges.asRanges().forEach(range -> {
+            String contains = mapOfRanges.get(range);
+            if (contains == null) {
+                rangesNotInHashMap.add(range);
+            }
+        });
+        assertThat(rangesNotInHashMap.size(), is(0));
+    }
+
+    private static List<Range<Long>> loadData(URL source, String dataRowIdentifier) {
+        Function<String[], Range<Long>> cardRangeExtractor = entry -> {
+            Long minCardDigit = BinRangeParser.calculateMinDigitForCardLength(Long.valueOf(entry[1]), CARD_RANGE_LENGTH);
+            Long maxCardDigit = BinRangeParser.calculateMaxDigitForCardLength(Long.valueOf(entry[2]), CARD_RANGE_LENGTH);
+            return Range.closed(minCardDigit, maxCardDigit);
```

Comment:
Hmm yeah it could be a private static method, I'll make the change.

Reply:
Although, given this is just test code, it doesn't really matter that we're instantiating this `cardRangeExtractor` on each thread, rather than placing it in memory reserved for private static fields.